src/core/qgsvectorfilewriter.cpp

```diff
@@ -362,12 +362,19 @@ void QgsVectorFileWriter::init( QString vectorFileName, QString fileEncoding, co
 
     switch ( attrField.type() )
     {
+#if defined(GDAL_VERSION_NUM) && GDAL_VERSION_NUM < 1700
       case QVariant::LongLong:
-        ogrType = OFTString;
-        ogrWidth = ogrWidth > 0 && ogrWidth <= 21 ? ogrWidth : 21;
-        ogrPrecision = -1;
+        ogrType = OFTInteger64;
```

Comment:
although - should this be for gdal < 2.0? I gather OFTInteger64 was added in 2.0 (https://trac.osgeo.org/gdal/wiki/rfc31_ogr_64)